models/models.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, models, fields, _
+import datetime as dt
+from odoo.exceptions import ValidationError
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+    active = fields.Boolean(default=True)
+    lifespan = fields.Integer(string='Lifespan')
+    USPhoneNumber = fields.Char(string="Phone Number")
+
+    def archive_purchase_order_form_view(self):
+        for rec in self:
+            rec.active = False

```

Comment:
What happens if we archive an order whose status is not in Done and Cancelled?
Also, try using `self.write({'active': False})` instead of this `for` loop